canard.c

```diff
@@ -878,6 +881,26 @@ CANARD_INTERNAL int16_t enqueueTxFrames(CanardInstance* ins,
     CANARD_ASSERT(ins != NULL);
     CANARD_ASSERT((can_id & CANARD_CAN_EXT_ID_MASK) == can_id);            // Flags must be cleared
 
+    uint8_t frame_max_length;
+
+    switch (protocol) {
+        case CanardTransportProtocolCAN:
+        {
+            frame_max_length = CANARD_CAN_FRAME_MAX_DATA_LEN;
+            break;
+        }
+        case CanardTransportProtocolFD:
+        {
+            frame_max_length = CANARD_FD_FRAME_MAX_DATA_LEN;
+            break;
+        }
+        default:
+        {
+            return -CANARD_ERROR_INVALID_ARGUMENT;
```

Comment:
How about adding `CANARD_ASSERT(false)` before the return statement? Also, the following break is unreachable, many an IDE would complain.

Reply:
My bad attempt at actually following MISRA, willfix

Reply:
MISRA is not self-contradictory here. It also says that there must be only one `return` statement per function, which can be difficult to follow; if you have to exit early, you may also be forced to break other rules, like having at least one `break` per case statement.